Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов.

Даны значения роста в трех группах случайно выбранных спортсменов:

Футболисты: 173, 175, 180, 178, 177, 185, 183, 182.

Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180.

Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.

In [1]:
import numpy as np

In [2]:
y1 = np.array([173, 175, 180, 178, 177, 185, 183, 182], dtype=np.float64)
y2 = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180], dtype=np.float64)
y3 = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170], dtype=np.float64)

In [17]:
n1 = len(y1)
n2 = len(y2)
n3 = len(y3)
n = n1 + n2 + n3
k = 3
n - k

25

In [24]:
y1_mean = np.mean(y1)
y2_mean = np.mean(y2)
y3_mean = np.mean(y3)
y_mean = (y1_mean * n1 + y2_mean * n2 + y3_mean * n3) / n
y1_mean, y2_mean, y3_mean, y_mean

(179.125, 178.66666666666666, 172.72727272727272, 176.46428571428572)

In [13]:
s2_f = ((y1_mean - y_mean)**2) * n1 + ((y2_mean - y_mean)**2) * n2 + ((y3_mean - y_mean)**2) * n3
sigma2_f = s2_f/ (k - 1)
s2_f, sigma2_f

(253.9074675324678, 126.9537337662339)

In [14]:
s2_r = np.sum((y1 - y1_mean)**2) + np.sum((y2 - y2_mean)**2) + np.sum((y3 - y3_mean)**2)
sigma2_r = s2_r/ (n - k)
s2_r, sigma2_r

(577.0568181818182, 23.08227272727273)

In [15]:
F_h = sigma2_f / sigma2_r
F_h

5.500053450812598

Табличное значение F для уровня значимости $\alpha$ = 0.05 и степеней свободы k-1 = 2 и n-k = 25 равно 3.38

Так как  5.5 > 3.38, различие роста в трех группах статистически значимо с уровнем значимости 0.05.

Табличное значение F для уровня значимости $\alpha$ = 0.01 и степеней свободы k-1 = 2 и n-k = 25 равно 5.72

Так как  5.5 < 5.72, различие роста в трех группах не является статистически значимым с уровнем значимости 0.01.

### Проверка

In [20]:
from scipy import stats

In [28]:
stats.f_oneway(y1, y2, y3)

F_onewayResult(statistic=5.500053450812596, pvalue=0.010482206918698694)

Pазличие роста в трех группах статистически значимо при уровне значимости 0.01048 и более

In [25]:
stats.f_oneway(y1, y2)

F_onewayResult(statistic=0.038473767885532584, pvalue=0.8471274829862326)

При этом различия роста у футболитов и хоккеистов статистически незначимы (при уровне значимости 0.847)

In [26]:
stats.f_oneway(y1, y3)

F_onewayResult(statistic=9.232797542249022, pvalue=0.007418502503470366)

In [27]:
stats.f_oneway(y2, y3)

F_onewayResult(statistic=6.86, pvalue=0.017384077963160147)

Раличия возрастов футболитов и штангистов статистически значимы на уровне значимости 0.0074,
а хоккеистов и штангистов - на уровне значимости 0.017.